cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/worker/RemoteWorkerPluginFinder.java

```diff
@@ -46,6 +46,6 @@ public class RemoteWorkerPluginFinder extends RemotePluginFinder {
   @Override
   protected Location getArtifactLocation(ArtifactId artifactId)
     throws IOException, ArtifactNotFoundException, UnauthorizedException {
-    return Locations.toLocation(artifactLocalizerClient.getUnpackedArtifactLocation(artifactId));
+    return Locations.toLocation(artifactLocalizerClient.getArtifactLocation(artifactId));
```

Comment:
So this is a performance regression since the caller needs to unpack everytime?

Reply:
Yes, we can perform optimizations per artifact, but I'm afraid that could get hacky. However, I've gone ahead and moved this to a separate PR since there are a couple of different methods for fixing the first bug.

Reply:
https://github.com/cdapio/cdap/pull/14458